codes/c/chapter_array_and_linkedlist/list.c

```diff
@@ -0,0 +1,187 @@
+/**
+ * File: list.c
```

Comment:
`list.c` is for introducing the built-in dynamic array. If there is no list in C, we should not provide it.

Also, you'd better not delete a merged code file, e.g., `my_list.c`